### PR TITLE
[symengine] Add version 0.13.0.

### PR DESCRIPTION
**Pull request by cqc-alec**
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by uilianries — COMMENTED** [review]
Thank you for your PR! 

This new minor includes fast float vendorized, not big deal, but we just need to check the licenses first: "https://github.com/symengine/symengine

**Comment by cqc-alec** [comment]
> Thank you for your PR!
> 
> This new minor includes fast float vendorized, not big deal, but we just need to check the licenses first: "https://github.com/symengine/symengine

Ah, this one? https://github.com/symengine/symengine/blob/0ccc20fea491ba0939768a2d74e590df2c1c2a74/LICENSE#L256

**Comment by cqc-alec** [comment]
Hi @uilianries , is there something I can do to help with this?

**Comment by AbrilRBS** [comment]
Hi @cqc-alec, Uilian is currently on vacaction, so I'm taking over his PRs for now

I've pushed some improvements to the recipe to modernize it. I've used the new `WITH_SYSTEM_FASTFLOAT` to ensure that it's not using the vendorized version, and made sure that gmp/boost also works correctly now (`-o="&:integer_class=boost"` used to fail compilation before this PR)

Thanks

**Comment by cqc-alec** [comment]
> Hi @cqc-alec, Uilian is currently on vacaction, so I'm taking over his PRs for now
> 
> I've pushed some improvements to the recipe to modernize it. I've used the new `WITH_SYSTEM_FASTFLOAT` to ensure that it's not using the vendorized version, and made sure that gmp/boost also works correctly now (`-o="&:integer_class=boost"` used to fail compilation before this PR)
> 
> Thanks

Thank you!

I also took the opportunity to update the boost version to 1.86.0.

File: recipes/symengine/all/test_package/conanfile.py [hunk 1]
```diff
@@ -17,7 +17,6 @@ def requirements(self):
 
     def generate(self):
         tc = CMakeToolchain(self)
-        tc.variables["CMAKE_CXX_STANDARD"] = "11"
```
Comment (cqc-alec) [inline recipes/symengine/all/test_package/conanfile.py:20]: Perhaps this is necessary to keep?

**Comment by uilianries** [comment]
Hello @cqc-alec! I'm currently on vacation :palm_tree:, please, ping `@conan-io/barbarians` for support. Have a nice day!

**Review by AbrilRBS — APPROVED** [review]
Looks good, sorry for the delay while we implemented the new CI system :)

**Review by uilianries — APPROVED** [review]
LGTM. https://github.com/symengine/symengine/compare/v0.12.0...v0.13.0#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR331

File: recipes/symengine/all/conanfile.py [hunk 2]
```diff
@@ -34,11 +39,47 @@ class SymengineConan(ConanFile):
...
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+    
+    def validate(self):
+        check_min_cppstd(self, self._min_cppstd)
+        
+        minimum_version = self._minimium_compilers_version.get(str(self.settings.compiler))
+        if minimum_version and Version(self.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires {self.settings.compiler} >= {minimum_version}")
```
Comment (cqc-alec) [inline recipes/symengine/all/conanfile.py:69]: Thank you for approving! I am now wondering if this is correct? When I do
```
conan install --requires symengine/0.13.0@ --build=missing
```
I get the error:
```
symengine/0.13.0: Invalid: symengine/0.13.0 requires gcc >= 7
```
But I have `compiler.version=13` in my settings. Is `self.version` the right thing to compare here?

Comment (cqc-alec) [reply recipes/symengine/all/conanfile.py:69]: https://github.com/conan-io/conan-center-index/pull/25884

**Pull request merged**